src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostMemberTyping.scala

```diff
@@ -28,28 +31,38 @@ trait GhostMemberTyping extends BaseTyping { this: TypeInfoImpl =>
         nonVariadicArguments(args)
   }
 
-  private[typing] def wellFoundedIfNeeded(member: PMember): Messages = {
-    val spec = member match {
-      case m: PMethodDecl => m.spec
-      case f: PFunctionDecl => f.spec
-      case _ => Violation.violation("Unexpected member type")
+  // spec must come from a ghost or pure function
+  private[typing] def wellFoundedIfNeeded(spec: PFunctionSpec): Messages = {
```

Comment:
three comments:
- can we change the behavior such that this function can be called for any spec and internally checks whether it's a spec for a ghost or pure function / method? That's the intuition I have for these `wellFoundedIf...` functions. Alternatively, please add a require stmt that checks this assumption
- Maybe I'm missing something but isn't this function only called for explicitly ghost functions and methods? To me, it seems like we do not call this function for actual pure functions and methods
- can we rename this function? Without looking at its implementation, I do not have any clue what `Needed` is supposed to do. What about `wellFoundedIfGhostOrPure`?